### ISSUE TITLE
[feature-request] style json `extends` fields

### ISSUE THREAD

**Issue by farfromrefug** [CartoDB/mobile-sdk#444]
In an effort to reduce style size and since most "sub-styles" (voyager, positron...) share the same json data it would nice to be able to use something like `"extends":"./voyager.json"` just like it works in most config files like `tsconfig.json`

Right now in my style it would reduce the style by 30kb

**Comment by mtehver** [comment]
Good idea, I have included this in develop branch (in mobile-carto-libs).

**Comment by farfromrefug** [comment]
@mtehver awesome! Didnot have time to test it but looking at the code the merge is done the wrong way. You are applying the extend to the original picojson. It should be the other way around.if extends it become the original and for every property in the "non extends" you overwrite the property from the "extends" picojson

**Comment by mtehver** [comment]
No, the file referenced by 'extends' field is taken as a base and fields from current file overwrite base fields. So it should work the same way as 'extends' in tsconfig.json works.

**Comment by farfromrefug** [comment]
Sorry my bad must have read it wrong !

**Comment by farfromrefug** [comment]
@mtehver saw one last thing which should be nice. Here https://github.com/CartoDB/mobile-carto-libs/blob/master/cartocss/src/cartocss/CartoCSSMapLoader.cpp#L166 when you "override" the extended style, if the "value" is an object it should be a "merge". This is what tsconfig does for `compilerOptions` for example. The really nice thing in our case is that it could allow to "override" or add some `nutiparameters`
